sklearn/metrics/tests/test_common.py

```diff
@@ -638,7 +638,8 @@ def test_inf_nan_input():
     for metric in CLASSIFICATION_METRICS.values():
         for y_true, y_score in invalids:
             assert_raise_message(ValueError,
-                                 "Can't handle mix of binary and continuous",
```

Comment:
Why would that defeat the purpose? If the input is nan or inf, that should be what's in the error message. If the input is a mix of binary and continuous, that should be in the error message.